{% extends 'base.html.twig' %}

{% block title %}Hello OrderController!{% endblock %}

{% block body %}
<style>
    .example-wrapper { margin: 1em auto; max-width: 800px; width: 95%; font: 18px/1.5 sans-serif; }
    .example-wrapper code { background: #F5F5F5; padding: 2px 6px; }
</style>

<div class="example-wrapper">
    <h1>Hello {{ controller_name }}! âœ…</h1>

    This friendly message is coming from:
    <ul>
        <li>Your controller at <code><a href="{{ 'C:/Users/anael/OneDrive/Bureau/Formation_CCI/Symfony/RetroShopSymfony/src/Controller/OrderController.php'|file_link(0) }}">src/Controller/OrderController.php</a></code></li>
        <li>Your template at <code><a href="{{ 'C:/Users/anael/OneDrive/Bureau/Formation_CCI/Symfony/RetroShopSymfony/templates/order/index.html.twig'|file_link(0) }}">templates/order/index.html.twig</a></code></li>
    </ul>
</div>
{% endblock %}
{% block script %}
    <script type="text/javascript">
        var stripe = Stripe("pk_test_51JebZVG4YTfuwl7ym7zRMIPz2W0PDaH2KcqzumGAQMV3zQsqRMA2wRYd8bJbEvSIueFusm7vECH3HdxPvEtYPrz500NRvM18sB");
        var checkoutButton = document.getElementById("checkout-button");
        checkoutButton.addEventListener("click", function () {
            fetch("/commande/create-session/{{ reference }}", {
                method: "POST",
            })
                .then(function (response) {
                    return response.json();
                })
                .then(function (session) {
                    if (session.error == 'order') {
                        window.location.replace('{{ path("order") }}');
                    } else {
                        return stripe.redirectToCheckout({ sessionId: session.id });
                    }
                })
                .then(function (result) {
                    // If redirectToCheckout fails due to a browser or network
                    // error, you should display the localized error message to your
                    // customer using error.message.
                    if (result.error) {
                        alert(result.error.message);
                    }
                })
                .catch(function (error) {
                    console.error("Error:", error);
                });
        });
    </script>
{% endblock %}